### PR TITLE
azure-pim-cli 0.6.1 (new formula)

### PR DESCRIPTION
**Pull request by tnaumann**
Confirming the following:

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

**Comment by carlocab** [comment]
Per the commit style guide,

> At Homebrew, we require the name of the formula up front like so: `foobar 7.3 (new formula).`

So your commit message should be
```
azure-pim-cli 0.6.1 (new formula)
```

**Comment by tnaumann** [comment]
@carlocab thank you! I've updated the commit message to reflect this

**Comment by carlocab** [comment]
The formula also doesn't pass `brew audit`. See the error log here: https://github.com/Homebrew/homebrew-core/actions/runs/11075398174/job/30776262556?pr=192113#step:5:28

You can try `brew audit --new --fix azure-pim-cli` to fix the auto-correctable errors, but there may be some that cannot be corrected automatically.

File: Formula/a/azure-pim-cli.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class AzurePimCli < Formula
+  desc "Unofficial CLI for Azure Privileged Identity Management (PIM) roles"
+  homepage "https://github.com/demoray/azure-pim-cli"
+  url "https://github.com/demoray/azure-pim-cli/archive/refs/tags/0.6.1.tar.gz"
+  sha256 "04f8157a5645d2c6e388b4bd151effda7545779c79dc537a2dd8dcba4d14379f"
+  license "MIT"
+  head "https://github.com/demoray/azure-pim-cli", branch: "main"
+
+  depends_on "rust" => :build
+  depends_on "azure-cli"
```
Comment (carlocab) [inline Formula/a/azure-pim-cli.rb:10]:
```suggestion
  depends_on "pkg-config" => :build
  depends_on "rust" => :build
  depends_on "azure-cli"

  on_linux do
    depends_on "openssl@3"
  end
```

**Review by carlocab — COMMENTED** [review]
I have a suggestion that will fix the Linux build, but I'm afraid there is a remaining audit failure that isn't as easy to fix:
```
  azure-pim-cli
    * HEAD: The source URL https://github.com/demoray/azure-pim-cli is not reachable
    * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
```

File: Formula/a/azure-pim-cli.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class AzurePimCli < Formula
+  desc "Unofficial CLI for Azure Privileged Identity Management (PIM) roles"
+  homepage "https://github.com/demoray/azure-pim-cli"
+  url "https://github.com/demoray/azure-pim-cli/archive/refs/tags/0.6.1.tar.gz"
+  sha256 "04f8157a5645d2c6e388b4bd151effda7545779c79dc537a2dd8dcba4d14379f"
+  license "MIT"
+  head "https://github.com/demoray/azure-pim-cli", branch: "main"
```
Comment (carlocab) [inline Formula/a/azure-pim-cli.rb:7]:
```suggestion
  head "https://github.com/demoray/azure-pim-cli.git", branch: "main"
```

File: Formula/a/azure-pim-cli.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  on_linux do
+    depends_on "openssl@3"
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/az-pim --version")
```
Comment (carlocab) [inline Formula/a/azure-pim-cli.rb:22]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Given the audit failure, it may be more suitable for you to host this formula in your own tap for now.

Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Pull request closed without merging**